google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -971,37 +983,46 @@ public long executeUpdate(Statement update) {
     Preconditions.checkNotNull(update);
     ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
     ParsedStatement parsedStatement = getStatementParser().parse(update);
-    if (parsedStatement.isUpdate()) {
-      switch (parsedStatement.getType()) {
-        case UPDATE:
+    switch (parsedStatement.getType()) {
+      case UPDATE:
+        if (this.autocommitDmlMode == AutocommitDmlMode.PARTITIONED_NON_ATOMIC) {
           return get(internalExecuteUpdateAsync(parsedStatement));
-        case CLIENT_SIDE:
-        case QUERY:
-        case DDL:
-        case UNKNOWN:
-        default:
-      }
+        }
+        ResultSet rs = internalExecuteQuery(parsedStatement, AnalyzeMode.NONE);
```

Comment:
* What happens if this is called while the connection is in read-only mode?
* What happens if this is called while a read-only transaction is active?

Reply:
Have added tests for these scenarios.